### PR TITLE
Add a new command: 'markdown-doc'

### PR DESCRIPTION
**Pull request by sebastiencs**
The command will return the markdown string of the
documentation, if available.

For [company-box](https://github.com/sebastiencs/company-box) I would like to render the documentation as markdown string, and for this I would need such command.

Before:
box-no-markdown
After:
box-markdown

### CODE REVIEW

File: company-lsp.el [hunk 1]
```diff
@@ -369,20 +369,25 @@ INCOMPLETE: t or nil. Whether the candidates are incomplete or not."
...
   "Get the documentation from the item in the CANDIDATE.
 
 The documentation can be either string or MarkupContent. This method
 will return markdown string if it is MarkupContent, original string
 otherwise. If the documentation is not present, it will return nil
-which company can handle."
+which company can handle.
+
+If ONLY-MARKDOWN is non-nil, it will return a documentation only if
+the string is a MarkupContent with `kind' set to `markdown'."
```
Comment (tigersoldier) [inline company-lsp.el]: nit: s/the string is a MarkupContent/the value is a hashtable representing MarkupContent/

MarkupContent is not a string :)

File: company-lsp.el [hunk 2]
```diff
@@ -369,20 +369,25 @@ INCOMPLETE: t or nil. Whether the candidates are incomplete or not."
...
          (documentation (gethash "documentation" item)))
-    (if
-        (hash-table-p documentation)  ;; If true, then the documentation is a MarkupContent. String otherwise.
-        (gethash "value" documentation)
-      documentation)))
+    (cond (only-markdown
+           (and (hash-table-p documentation)
+                (equal (gethash "kind" documentation) "markdown")
+                (gethash "value" documentation)))
+          ((hash-table-p documentation) (gethash "value" documentation))
```
Comment (tigersoldier) [inline company-lsp.el]: nit: keep the documentation that documentation being a hashtable implies it's a MarkupContent

**Comment by dgutov** [comment]
I'll ask here as well: any reason why `doc-buffer` doesn't return a buffer with contents formatted as such already?

**Comment by sebastiencs** [comment]
Sorry for the delay @dgutov, you're right it would be better if using directly `doc-buffer`.
@tigersoldier Would you accept such changes ? (which makes `markdown-mode` a dependency)

**Comment by yyoncho** [comment]
@sebastiencs it could be optional. There are methods handling rendering in lsp-mode - check lsp--render-string  .

**Comment by sebastiencs** [comment]
@yyoncho Alright, I will make a new PR

**Pull request closed without merging**